rfcs/0027-formalize-documentation-organization.md

```diff
@@ -0,0 +1,442 @@
+should tell the user what problem the how-to is solving.
+
+How are tutorials different from how-tos? A tutorial is oriented towards the
+new developer, and focuses on successfully introducing them to the software and
+community. A how-to in contrast focuses on accomplishing a specific task within
+the context of basic understanding. A tutorial helps to onboard, a how-to helps
+to accomplish a task.
+
+### Reference
+
+Reference documentation describes how the software is configured and operated.
+APIs, key functions, commands, and interfaces are all candidates for reference
+documentation. These are the technical manuals that let users build their own
+interfaces and programs. They are information oriented, focused on lists and
+descriptions. You can assume that the audience has a grasp on how the software
+works and is looking for specific answers to specific questions. Ideally, the
+reference documentation should have the same structure as the code base and
+generated automatically as much as possible.
+
+### Explanations (Deep Dive)
+
+Background material on a topic. These documents help to illuminate and
+understand the application environment. Why are things the way they are? What
+were the design decisions, what alternatives were considered, what are the RFCs
+describing the existing system. This includes academic papers and links to
+publications relevant to the software. Within these documents you can explore
+contradictory and conflicting position, and help the reader make sense of how
+and why the software was built the way it is. It’s not the place for how-to’s
+and technical descriptions, and instead focuses on higher level concepts.
+
+## Special considerations for TVM
+
+The TVM comunity has some special considerations that require deviation from
+the simple docs style outlined by Divio. The first is that there is frequently
+overlap between the user and developer communities. Many projects document the
+develper and user experience with separate systems, but it is appropriate to
+consider both in this system, with differentiations where appropriate. As a
+result the tutorials and how-tos will be divided between "User Guides" and
+"How-to Guides".
+
+The next is that there are special topics within the TVM community that benefit
+from additional attention. These topics include, but are not limited to,
+microTVM and VTA. Special "Topic Guides" can be created to index existing
+material, and provide context on how to naviagate that material most
+effectively.
+
+To facilitate newcomers, a special "Getting Started" section with installation
+instructions, a overview of why to use TVM, and other first-experience douments
+will be produced. 
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Document Organization
+
+### Top Level Organization
+
+* Getting Started
+* User Guide
+* Topic Guide
+* Developer Guide
+* TVM Architecture Guide
+* API Reference (reference)
```

Comment:
minor note: we have language reference which served as a comprehensive reference to the language specification. see https://tvm.apache.org/docs/langref/index.html, would be great to clarify where do they fit

Reply:
It could go either way, in the Architecture Guide or in the Reference. Since it started as a reference document, and language references are much more status, I expanded the Reference section to include API and Language. Let me know what you think.